### PR TITLE
[2.11.x] DDF-3416 Added OutOfMemory and VM error support on Windows

### PR DESCRIPTION
**Pull request by paouelle**
**Master PR: [2615](https://github.com/codice/ddf/pull/2615)**

**Do not squash the 2 commits**

#### What does this PR do?
This PR adds support to Windows for automatically restarting ddf when a fatal error is detected and when an out of memory error is detected. It also fixes the restart when ddf is started via the service wrapper.
#### Who is reviewing it? 
@figliold 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@clockard
@coyotesqrl
@lessarderic
#### How should this be tested? (List steps with links to updated documentation)
1. Unzip and install the distribution on Windows
2. Go through the installation and make sure that the restart step at the end still works
3. Force an out of memory (see below). The process should restart
4. run a migration:export followed by a migration:import, the process should restart automatically
5. Go through the documentation to install the service wrapper
6. re-do step #2 and #4 via the client.bat tool

The only way to verify the out of memory is to manually modify the code somewhere to create a leak. Something like this:
```final List list = new ArrayList();

      try {
        while (true) {
          list.add(new int[Integer.MAX_VALUE]);
        }
      } catch (OutOfMemoryError e) { // swallow
      }
```
in the code of a command that you can trigger from the shell would do the trick.
#### What are the relevant tickets?
[DDF-3416](https://codice.atlassian.net/browse/DDF-3416)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by paouelle** [comment]
build now

File: distribution/ddf-common/src/main/resources/bin/ddf.bat [hunk 1]
```diff
@@ -1,9 +1,27 @@
...
+  DEL "%DIRNAME%\restart.jvm"
+)
 
+REM Actually invoke ddf to gain restart support
+CALL "%DIRNAME%\karaf.bat" %ARGS%
+SET RC=%ERRORLEVEL%
+
+IF "%1"== "daemon" (
+  EXIT /B %RC%
+) else (
```
Comment (figliold) [inline distribution/ddf-common/src/main/resources/bin/ddf.bat]: ✏️ Shouldn't this say ddf_on_error.bat?

File: distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf [hunk 2]
```diff
@@ -29,17 +29,16 @@ wrapper.java.additional.22 = -Djava.awt.headless=true
...
 
 # The Security Manager is turned off by default in DDF.
 # N.B. The use of the double equals on the 'java.security.policy' property is intentional.
 # See http://docs.oracle.com/javase/7/docs/technotes/guides/security/PolicyFiles.html#DefaultLocs
 # for more information.
-# wrapper.java.additional.24 = -Djava.security.policy==%KARAF_HOME%/security/default.policy
-# wrapper.java.additional.25 = -Dpolicy.provider=net.sourceforge.prograde.policy.ProGradePolicy
-# wrapper.java.additional.26 = -DproGrade.getPermissions.override=sun.rmi.server.LoaderHandler:loadClass,org.apache.jasper.compiler.JspRuntimeContext:initSecurity
+# wrapper.java.additional.26 = -Djava.security.policy==%KARAF_HOME%/security/default.policy
+# wrapper.java.additional.27 = -Dpolicy.provider=net.sourceforge.prograde.policy.ProGradePolicy
```
Comment (figliold) [inline distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf]: ❓ Is there something wrong with the formatting here?

**Comment by paouelle** [comment]
build now

**Comment by paouelle** [comment]
build now

**Comment by paouelle** [comment]
build now

**Pull request merged**